iterative/utils/analytics.go

```diff
@@ -0,0 +1,290 @@
+	cmd := exec.Command("terraform", "--version")
+	cmd.Stdout = &out
+
+	err := cmd.Run()
+
+	if err == nil {
+		regex := regexp.MustCompile(`v\d+\.\d+\.\d+`)
+		version := regex.FindString(out.String())
+		return version
+	} else {
+		logrus.Error("Analytics: Failed extracting terraform version")
+	}
+
+	return ""
+}
+
+func GroupId() string {
+	if !IsCI() {
+		return ""
+	}
+
+	id, err := deterministic(fmt.Sprintf("%s%s%s%s%s",
+		os.Getenv("GITHUB_SERVER_URL"),
+		os.Getenv("GITHUB_REPOSITORY_OWNER"),
+		os.Getenv("CI_SERVER_URL"),
+		os.Getenv("CI_PROJECT_ROOT_NAMESPACE"),
+		os.Getenv("BITBUCKET_WORKSPACE"),
+	))
+
+	if err != nil {
+		return ""
+	}
+
+	return id.String()
+}
+
+func UserId() string {
+	id := uuid.New().String()
+	old := appdirs.UserConfigDir("dvc/user_id", "iterative", "", false)
+	new := appdirs.UserConfigDir("telemetry", "iterative", "", false)
+
+	_, errorOld := os.Stat(old)
+	if !os.IsNotExist(errorOld) {
+		os.Rename(old, new)
+	}
+
+	_, errorNew := os.Stat(new)
+	if !os.IsNotExist(errorNew) {
+		dat, _ := ioutil.ReadFile(new)
+		id = string(dat[:])
+	} else {
+		ioutil.WriteFile(new, []byte(id), 0644)
+	}
+
+	if IsCI() {
+		uid, err := deterministic(fmt.Sprintf("%s%s%s",
+			os.Getenv("GITHUB_ACTOR"),
+			os.Getenv("GITLAB_USER_ID"),
+			os.Getenv("BITBUCKET_STEP_TRIGGERER_UUID"),
+		))
+
+		if err == nil {
+			id = uid.String()
+		}
+	}
+
+	return id
+}
+
+func ResourceData(d *schema.ResourceData) map[string]interface{} {
+	if d == nil {
+		return map[string]interface{}{}
+	}
+
+	tpiLogs := d.Get("logs").([]interface{})
+	logs := ""
+	for _, log := range tpiLogs {
+		logs += log.(string)
+	}
+	spot := d.Get("spot").(float64)
+	return map[string]interface{}{
+		"cloud":           d.Get("cloud").(string),
+		"cloud_region":    d.Get("region").(string),
+		"cloud_machine":   d.Get("machine").(string),
+		"cloud_disk_size": d.Get("disk_size").(int),
+		"cloud_spot":      spot,
+		"cloud_spot_auto": spot == 0.0,
+		"task_status":     d.Get("status").(map[string]interface{}),
+		"task_duration":   TaskDuration(logs),
+		"task_resumed":    len(tpiLogs) > 1,
+	}
+}
+
+func JitsuEventPayload(action string, e error, extra map[string]interface{}) map[string]interface{} {
+	systemInfo := SystemInfo()
+
+	extra["ci"] = guessCI()
+	extra["terraform_version"] = TerraformVersion()
+
+	err := ""
+	if e != nil {
+		err = reflect.TypeOf(e).String()
+	}
+
+	payload := map[string]interface{}{
+		"user_id":      UserId(),
+		"group_id":     GroupId(),
+		"action":       action,
+		"interface":    "cli",
+		"tool_name":    "tpi",
+		"tool_source":  "terraform",
+		"tool_version": Version,
+		"os_name":      systemInfo["platform"],
+		"os_version":   systemInfo["platform_version"],
+		"backend":      extra["cloud"],
+		"error":        err,
+		"extra":        extra,
+	}
+
+	return payload
+}
+
+func SendJitsuEvent(action string, e error, extra map[string]interface{}) {
+	for _, env := range []string{"DO_NOT_TRACK", "DVC_NO_ANALYTICS"} {
```

Comment:
There are some [discussions](https://iterativeai.slack.com/archives/C01APS0FHDM/p1629152270013500) regarding the adopition of `DO_NOT_TRACK` we should revisit before including it.

Reply:
Can we unblock those discussions?

Reply:
Yes, let me ping people [here](https://iterativeai.slack.com/archives/C6YHPP2TB/p1652206667050159?thread_ts=1652196801.349989&cid=C6YHPP2TB).

Reply:
Created a thread [here](https://iterativeai.slack.com/archives/C0209RF6A5B/p1652345394951249)

Reply:
Maybe we can remove it for now and if we want to include in the future everything will be OK
At least we can unblock this